vignettes/common_structure.Rmd

```diff
@@ -0,0 +1,82 @@
+---
+title: "Common Structure"
+output: 
+  rmarkdown::html_vignette
+vignette: >
+  %\VignetteIndexEntry{Common Structure}
+  %\VignetteEngine{knitr::rmarkdown}
+  %\VignetteEncoding{UTF-8}
+---
+
+# Common structure of the workflows
+
+## R-version input
+
+For most of the workflows, we need to select a specific R version to load the associated `renv` profile.
+That's why most of the actions have a `r-version` input (by default set-up to `4.1`) :
+
+```
+inputs:
+    r-version:
+    description: 'The version of R to use'
+    default: '4.1'
+    required: false
+    type: string
+```
+
+## Use of `admiralci` docker images
+
+Once the `r-version` is picked, the worfklows will run inside a docker container instance. Indeed, `admiralci` contains an action `push-docker-image`, to be able to build `rocker:rstudio` images
+for each available versions of `R` (versions specified in `renv/profiles`, see [renv profiles conf](get_started.html#renv-profiles-configuration-handling-several-r-versions)). 
+At the building step of these docker images, every dependencies specified in the `renv.lock` files are installed.
+A total of 4 images are created :
+- `admiralci-4.1`
+- `admiralci-4.2`
+- `admiralci-4.3`
+- `admiralci-devel`
+
+These images and store them in admiralci ghcr (Github Container Registry - Place to store docker images), and are then reused to execute the workflows in docker container instances like this :
+
+```
+container:
+    image: "ghcr.io/pharmaverse/admiralci-${{ inputs.r-version }}:latest"
+```
+
+(More more details on docker images creation process and how they're used, please read [Docker images and codespaces](docker_and_codespaces.html)) section.
+
+## Checkout repository
+
+This action is used to checkout on the right repository. This is a step commonly used in github actions to be able to access a repository tree.
+
+```
+- name: Checkout repository
+  uses: actions/checkout@v3
+```
+
+## Staged dependencies
+
+The staged dependencies action allow to install every necessary upstream dependencies for `admiral` package, suche as :
```

Comment:
typo: "dependencies" instead of "dependency"

Reply:
also "suche" instead of "such"

Reply:
also missing full stops.

Reply:
```suggestion
The staged dependencies action allow to install every necessary upstream dependencies for `admiral` package, such as :
```